File: apps/server/src/shared/utils/token.js
import crypto from "crypto";
import { createHash } from "crypto";

const generateRefreshToken = () => {
  return crypto.randomBytes(40).toString("hex");
};

const generateVerificationToken = () => {
  return crypto.randomBytes(40).toString("hex");
};

const hashToken = (token) => createHash("sha256").update(token).digest("hex");

export { generateRefreshToken, generateVerificationToken, hashToken };

File: apps/server/app/shared/utils/token.py
import hashlib
import secrets


def generate_refresh_token() -> str:
    return secrets.token_hex(40)


def generate_verification_token() -> str:
    return secrets.token_hex(40)


def hash_token(token: str) -> str:
    return hashlib.sha256(token.encode("utf-8")).hexdigest()
